src/storage-controller/src/lib.rs

```diff
@@ -1970,6 +2029,18 @@ where
             .map(|(id, c)| (*id, c))
     }
 
+    /// Iterate over exports that have not been dropped.
+    fn active_exports(&self) -> impl Iterator<Item = (GlobalId, &ExportState<T>)> {
+        // When an export is dropped it is removed from `sink_statistics`.
+        // This is currently the only indication of dropped exports.
```

Comment:
LMK if this is not true!

Reply:
it kindof is, but its not really something we should rely on 

i think the right thing to do here is add a `dropped` field to `ExportState` and set it at the same time as when remove the sink statistics (so, when we process `pending_sink_drops`)

I actually don't know why we don't just remove the `export` when we drop sink; this may be a bug, as it holds onto a capability forever; @sploiselle do you know?

Reply:
There is this comment further up which gives a hint:

```
    /// Collections maintained by the storage controller.
    ///
    /// This collection only grows, although individual collections may be rendered unusable.
    /// This is to prevent the re-binding of identifiers to other descriptions.
```

I don't know enough to say how big the danger of rebinding is here. The compute controller certainly doesn't have this kind of protection, but it also would become very confused if adapter tried to reuse an ID.

Reply:
Happy to add a `dropped` flag, but let's wait what @sploiselle thinks.

FWIW, for `collections` we check whether one was dropped like this:

```rust
    pub fn is_dropped(&self) -> bool {
        self.read_capabilities.is_empty()
    }
```

Seems reasonable to have an equivalent check for exports too, if we can set `ExportState::read_capability` to the empty frontier when an export gets dropped.

Reply:
I am under the impression that the only reason these remain in memory is, as the comment says, to prevent re-binding.

Orthogonal to that point, though, when we drop a sink, we should also release its capability by advancing the `ExportState`'s `write_frontier` to the empty antichain, which should, in turn, downgrade the read capability to the empty antichain, as well--which _then_ releases the read capability on its inputs.

So:

>  if we can set ExportState::read_capability to the empty frontier when an export gets dropped.

should already occur. That is also to say that:

        // When an export is dropped it is removed from `sink_statistics`.
        // This is currently the only indication of dropped exports.

is not true.

Have we seen cases where this didn't happen?

Reply:
> Have we seen cases where this didn't happen?

Nope! I simply wasn't aware about these invariants wrt. dropping of exports.

I added an `ExportState::is_dropped` method now that checks for an empty `read_capability`, similar to what we already do for collections.